client/login/magic-login/handle-emailed-link-form-jetpack-connect.tsx

```diff
@@ -0,0 +1,125 @@
+	isTwoFactorEnabled,
+} from 'calypso/state/login/selectors';
+import { recordTracksEventWithClientId as recordTracksEvent } from 'calypso/state/analytics/actions';
+
+interface Props {
+	emailAddress: string;
+	token: string;
+}
+
+const HandleEmailedLinkFormJetpackConnect: FC< Props > = ( { emailAddress, token } ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const [ hasSubmitted, setHasSubmitted ] = useState( false );
+
+	const redirectToOriginal = useSelector( ( state ) => getRedirectToOriginal( state ) || '' );
+	const redirectToSanitized = useSelector( ( state ) => getRedirectToSanitized( state ) );
+	const authError = useSelector( ( state ) => getMagicLoginRequestAuthError( state ) );
+	const isAuthenticated = useSelector( ( state ) =>
+		getMagicLoginRequestedAuthSuccessfully( state )
+	);
+	const isExpired = useSelector(
+		( state ) => getMagicLoginCurrentView( state ) === LINK_EXPIRED_PAGE
+	);
+	const isFetching = useSelector( ( state ) => isFetchingMagicLoginAuth( state ) );
+	const twoFactorEnabled = useSelector( ( state ) => isTwoFactorEnabled( state ) );
+	const twoFactorNotificationSent = useSelector( ( state ) =>
+		getTwoFactorNotificationSent( state )
+	);
+
+	useEffect( () => {
+		if ( isEmpty( emailAddress ) || isEmpty( token ) ) {
+			dispatch( showMagicLoginLinkExpiredPage() );
+		} else {
+			setHasSubmitted( true );
+			dispatch( fetchMagicLoginAuthenticate( token, redirectToOriginal ) );
+		}
+	}, [] );
+
+	// Lifted from `blocks/login`
+	// @TODO move to `state/login/actions` & use both places
+	const handleValidToken = useCallback( () => {
+		if ( ! twoFactorEnabled ) {
+			dispatch( rebootAfterLogin( { magic_login: 1 } ) );
+		} else {
+			page(
+				login( {
+					isNative: true,
+					// If no notification is sent, the user is using the authenticator for 2FA by default
+					twoFactorAuthType: twoFactorNotificationSent.replace( 'none', 'authenticator' ),
+					redirectTo: redirectToSanitized,
+				} )
+			);
+		}
+	}, [ dispatch ] );
+
+	useEffect( () => {
+		if ( ! hasSubmitted || isFetching ) {
+			// Don't do anything here unless the browser has received the `POST` response
+			return;
+		}
+
+		if ( authError || ! isAuthenticated ) {
+			// @TODO if this is a 5XX, or timeout, show an error...?
+			dispatch( showMagicLoginLinkExpiredPage() );
+			return;
+		}
+
+		handleValidToken();
+	}, [ authError, dispatch, handleValidToken, hasSubmitted, isAuthenticated, isFetching ] );
+
+	if ( isExpired ) {
+		return <EmailedLoginLinkExpired />;
+	}
+
+	const illustration = '/calypso/images/illustrations/' + 'illustration-nosites.svg';
+
+	dispatch( recordTracksEvent( 'calypso_login_email_link_handle_click_view' ) );
+
+	return (
+		<EmptyContent
+			className={ classNames( 'magic-login__handle-link', {
+				'magic-login__is-fetching-auth': isFetching,
+			} ) }
+			illustration={ illustration }
+			illustrationWidth={ 500 }
+			title={ translate( 'Logging in as %(emailAddress)s …', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Logging in as %(emailAddress)s' )` *ES Score: 9*